oracle/src/pasta/fp_kimchi.rs

```diff
@@ -0,0 +1,828 @@
+use crate::poseidon::ArithmeticSpongeParams;
```

Comment:
so IIUC these parameters were missing before? How was anything between ocaml and rust working then?

Reply:
They previously had a different name, but they were present before. I believe @jspada is planning to create a Mina PR and a snarky PR with corresponding changes before merging this, so that everything keeps in sync

Reply:
I'm confused :o it looks like we were using the now {fp, fq}_legacy, whereas after this PR we use {fp, fq}_kimchi which are brand new in this repo

Reply:
They were present under another name, but unused in proof-systems, which as far and Matthew and I know was a mistake.

This PR adds them back and gives them better names so they cannot be confused again, here nor anywhere else.  It also lines up the names of the params with the sponge constant names, so that it's more obvious which combos are valid. I'll be opening up mina and snarky PRs corresponding to this also.